cmd/zetae2e/local/local.go

```diff
@@ -492,6 +491,13 @@ func localE2ETest(cmd *cobra.Command, _ []string) {
 
 	logger.Print("✅ e2e tests completed in %s", time.Since(testStartTime).String())
 
+	if testSolana {
+		require.True(
+			deployerRunner,
+			deployerRunner.VerifySolanaContractsUpgrade(conf.AdditionalAccounts.UserSolana.SolanaPrivateKey.String()),
+		)
+	}
+
```

Comment:
So we don't run any tests after the upgrade?

Reply:
We run it at the end of a test; that should be enough to test everything we need to assert.
I am not sure how node binary upgrades and program upgrade are related , lmk if there is a specific condition you are thinking about

Reply:
We don't seem to do anything to validate that the new program actually works?

Reply:
No, but we assert the state logic.
Testing the contract itself should be a separate test; this test is for upgrading and making sure the upgrade works.

I think a better plan would be just to include solana tests into our upgrade test , which we should be doing anyway , which would be able to test the new contract , but that's out of scope for this PR